### PR TITLE
kinetis: move filtering-out periph_hwrng in cpu/kinetis

### PR DESCRIPTION
**Pull request by cladmi**
### Contribution description

This removes doing `filter-out periph_hwrng, $(FEATURES_PROVIDED)`
after processing `cpu/$(CPU)/Makefile.features`.
The current solution is a HACK as `CPU_MODEL` is currently not available
at that moment but will be in the near future.

It will allow always including `cpu/$(CPU)/Makefile.features` after
`boards/$(BOARD)/Makefile.features`.

It is a part of moving `CPU/CPU_MODEL` definitions to `Makefile.features`.

### Testing procedure

For all the modified boards, the provided features are the same as in `master`.
I only checked `examples/hello-world` as it should not change anything for these ones.

```
for board in frdm-k64f frdm-kw41z phynode-kw41z teensy31 usb-kw41z; do echo ${board}; BOARD=${board} make --no-print-directory -C examples/hello-world info-debug-variable-FEATURES_PROVIDED; done
frdm-k64f
periph_adc periph_i2c periph_pwm periph_rtc periph_rtt periph_spi periph_timer periph_uart periph_cpuid periph_gpio periph_gpio_irq periph_mcg periph_pm cpp
frdm-kw41z
periph_adc periph_i2c periph_rtc periph_rtt periph_spi periph_timer periph_uart periph_cpuid periph_gpio periph_gpio_irq periph_mcg periph_pm cpp
phynode-kw41z
periph_adc periph_i2c periph_rtc periph_rtt periph_spi periph_timer periph_uart periph_cpuid periph_gpio periph_gpio_irq periph_mcg periph_pm cpp
teensy31
periph_pwm periph_rtc periph_rtt periph_timer periph_uart periph_cpuid periph_gpio periph_gpio_irq periph_mcg periph_pm cpp
usb-kw41z
periph_adc periph_i2c periph_rtc periph_rtt periph_spi periph_timer periph_uart periph_cpuid periph_gpio periph_gpio_irq periph_mcg periph_pm cpp
```

I also looked at all boards to verify they did not change but did not post the output.

### Issues/PRs references

Required to apply https://github.com/RIOT-OS/RIOT/pull/11478
Split out of https://github.com/RIOT-OS/RIOT/pull/11477

### CODE REVIEW

File: cpu/kinetis/Makefile.features [hunk 1]
```diff
@@ -1,5 +1,18 @@
 FEATURES_PROVIDED += periph_cpuid
-FEATURES_PROVIDED += periph_hwrng
+
+# HACK Do not define 'hwrng' if the board does not supports it
+# I prefer a whitelist on CPU_MODEL but the information is not available yet
```
Comment (aabadie) [inline cpu/kinetis/Makefile.features]: This comment has too much personal taste (e.g. don't use  `I prefer`). Maybe reword with: `A whitelist on CPU_MODEL would be better but this information/variable is not available yet.`

File: cpu/kinetis/Makefile.features [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+# HACK the board/cpu_model currently uses the wrong hwrng register
```
Comment (aabadie) [inline cpu/kinetis/Makefile.features]: If I understand this correctly, the wrong hwrng register is used for frdm-k64f ? Maybe make this more explicit that this HACK applies to frdm-kw64f.

Comment (cladmi) [reply cpu/kinetis/Makefile.features]: I forgot to re-precise this when moving it to the common directory.

File: cpu/kinetis/Makefile.features (hunk 1 above)
Comment (cladmi) [reply cpu/kinetis/Makefile.features]: Indeed, will change.

File: cpu/kinetis/Makefile.features (hunk 2 above)
Comment (cladmi) [reply cpu/kinetis/Makefile.features]: While re-reading I am not sure if it is the `frdm-k64f` or the `CPU_MODEL` completely. That is why I wrote `board/cpu_model`. I think it may be more cpu_model but hard to know without debugging why.

I can change to `Hack, either the board or the cpu_model currently uses the wrong hwrng` maybe ?

Comment (aabadie) [reply cpu/kinetis/Makefile.features]: Maybe start the next sentence on the same line:
`HACK the board/cpu_model currently uses the wrong hwrng register. Remove next line when fixed` (you can split it if it's too long)

Comment (cladmi) [reply cpu/kinetis/Makefile.features]: I can just remove the "remove next line when fixed" as it is implicit and still precise the `frdm` as it is currently the only one I know as you said in the first comment.
`The HWRNG uses the wrong hwrng register for the frdm-k64f board/cpu_model`

**Comment by cladmi** [comment]
Squashed.

**Comment by cladmi** [comment]
Thanks :)
